libr/bin/bin.c

```diff
@@ -1010,17 +1007,12 @@ static RBinObject *r_bin_object_new(RBinFile *binfile, RBinPlugin *plugin, ut64
 		return NULL;
 	}
 
-	o->plugin = plugin;
-	o->loadaddr = loadaddr;
-	o->baddr = baseaddr;
-	o->baddr_shift = 0;
 	// XXX - binfile could be null here meaning an improper load
 	// XXX - object size cant be set here and needs to be set where
 	// where the object is created from.  The reason for this is to prevent
 	// mis-reporting when the file is loaded from impartial bytes or is extracted
 	// from a set of bytes in the file
 	r_bin_object_set_items (binfile, o);
-	r_bin_file_object_add (binfile, o);
```

Comment:
ok this is causing the issue why any binary can be loaded, why did you remove this line?